lib/client/bespoke-client.ts

```diff
@@ -72,13 +73,32 @@ export class BespokeClient {
     }
 
     private onWebhookReceived(request: WebhookRequest): void {
-        let self = this;
+        const self = this;
+
+        if (this.secretKey) {
+            let secretKeyValidated: boolean = false;
+
+            if (request.headers && request.headers["secretkey"] === this.secretKey) {
```

Comment:
I think this should be labeled bespoken-key - secretkey is too generic, and also draws attention to itself